cypress/e2e/tests/pages/user_menu/preferences.spec.ts

```diff
@@ -0,0 +1,368 @@
+    */
+    HomePagePo.goTo();
+    userMenu.open();
+    userMenu.checkOpen();
+    userMenu.clickMenuLink('Preferences');
+    userMenu.checkClosed();
+    cy.url().should('include', endpoint);
+    prefPage.title();
+  });
+
+  it('Can select a language', () => {
+    /*
+    Select language and verify its preserved after page reload
+    */
+    const languages = {
+      简体中文:    '[lang="zh-hans"]',
+      English: '[lang="en-us"]'
+    };
+
+    PagePo.goTo(endpoint);
+    for (const [key, value] of Object.entries(languages)) {
+      prefPage.dropdownMenu().open(0);
+      prefPage.listBox().isOpened();
+      prefPage.listBox().getListBoxItems().should('have.length', 2);
+      prefPage.listBox().set(key);
+      prefPage.dropdownMenu().checkOptionSelected(0, key);
+      prefPage.listBox().isClosed();
+      prefPage.checkLangDomElement(value);
+      cy.reload();
+      prefPage.dropdownMenu().checkOptionSelected(0, key);
+      prefPage.checkLangDomElement(value);
+    }
+  });
+
+  it('Can select a theme', () => {
+    /*
+    Select theme and verify that its highlighted
+    Verify selection and mode ('Light' or 'Dark') is preserved after logout/login
+    */
+    const themeOptions = {
+      Light: 'theme-light',
+      Dark:  'theme-dark',
+      Auto:  ''
+    };
+    // Set theme for 'Auto' mode based on time of day
+    const hour = new Date().getHours();
+
+    for (const key in themeOptions) {
+      (key === 'Auto' && hour < 7 || hour >= 18) ? themeOptions['Auto'] = 'theme-dark' : themeOptions['Auto'] = 'theme-light';
+    }
+
+    PagePo.goTo(endpoint);
+    for (const [key, value] of Object.entries(themeOptions)) {
+      PagePo.goTo(endpoint);
+      prefPage.button(key).set();
+      prefPage.button(key).isSelected();
+      prefPage.checkThemeDomElement(value);
+      userMenu.open();
+      userMenu.checkOpen();
+      userMenu.clickMenuLink('Log Out');
+      cy.login();
+      PagePo.goTo(endpoint);
+      prefPage.button(key).isSelected();
+      prefPage.checkThemeDomElement(value);
+    }
+  });
+
+  it('Can select login landing page', () => {
+    /*
+    Select each radio button and verify its highlighted
+    Verify user is landing on correct page after login
+    Verify selection is preserved after logout/login
+    */
+    const landingPageOptions: {[key: number]: string} = {
+      0: '/home',
+      1: 'c/_/manager/provisioning.cattle.io.cluster',
+      // 2: '/explore' // TODO this option only works when there is an existing cluster (not for standard user)
+    };
+
+    for (const [key, value] of Object.entries(landingPageOptions)) {
+      PagePo.goTo(endpoint);
+      prefPage.radioButton().set(key);
+      prefPage.radioButton().isChecked(key);
+
+      // if key is 1, navigate to cluster manager page and then do validations, else just do validations
+      if (key == 1) {
+        cy.intercept('GET', '/v3/clusters').as('clusters');
+        PagePo.goTo('c/_/manager/provisioning.cattle.io.cluster');
```

Comment:
This wouldn't work because the url in ClusterManagerListPagePo uses a static url
`/c/local/manager/provisioning.cattle.io.cluster`

the url I am trying to reach is:
`c/_/manager/provisioning.cattle.io.cluster`

We would need to update the class to accept an argument for `clusterId`

Reply:
updated: https://github.com/rancher/dashboard/pull/8705/commits/7b053833f6b23453ea1ba49cd7ef26418bd448c0

Refactored ClusterManagerListPagePo to take clusterId as argument and updated spec files used that class